lib/onelogin/ruby-saml/authrequest.rb

```diff
@@ -46,11 +64,13 @@ def create_params(settings, params={})
 
         if settings.security[:authn_requests_signed] && !settings.security[:embed_sign] && settings.private_key
           params['SigAlg']    = settings.security[:signature_method]
-          url_string          = "SAMLRequest=#{CGI.escape(base64_request)}"
-          url_string         << "&RelayState=#{CGI.escape(relay_state)}" if relay_state
-          url_string         << "&SigAlg=#{CGI.escape(params['SigAlg'])}"
-          private_key         = settings.get_sp_key()
-          signature           = private_key.sign(XMLSecurity::BaseDocument.new.algorithm(settings.security[:signature_method]).new, url_string)
+          url_string = OneLogin::RubySaml::Utils.build_query(
+            :type        => 'SAMLRequest',
+            :data        => base64_request,
+            :relay_state => relay_state,
+            :sig_alg     => params['SigAlg']
+          )
+          signature           = settings.get_sp_key().sign(XMLSecurity::BaseDocument.new.algorithm(settings.security[:signature_method]).new, url_string)
```

Comment:
Unnecessary parens here. Also, remove the extra whitespace before `=`, and assign the xml security algorithm to a variable on the line above and use it here _(vs. inlining the call)_ for readability.